primeqa/components/retriever/dense.py

```diff
@@ -138,15 +139,26 @@ def predict(self, input_texts: List[str], *args, **kwargs) -> Any:
             else self.max_num_documents
         )
 
-        # TODO: Add kwarg defining return format (List[List[Tuple(pids, score)]], List[List[<document>]])
-        ranking_results = self._searcher.search_all(
-            {idx: str(input_text) for idx, input_text in enumerate(input_texts)},
-            k=max_num_documents,
+        passage_ids=[]
+        scores=[]
+        p_ids, response = self._searcher.search(
```

Comment:
Is this correct?
It looks to me that `search` expects a single text string here: https://github.com/primeqa/primeqa/blob/main/primeqa/ir/dense/colbert_top/colbert/searcher.py#L100

Reply:
@franzmpub : this points to https://github.com/primeqa/primeqa/blob/main/primeqa/ir/dense/dpr_top/dpr/searcher.py#L134 - which supports a mode for query list

Reply:
@jdpsen Are you sure?  I see the the DPRRetriever class starting on line 165, this is still in the ColBERTRetriever class as far as I can tell.

Reply:
@jdpsen : I still see a call to `self._searcher.search` with multiple strings (`query_batch=input_texts`) as the first argument.
Why is this change in `ColBERTRetriever` needed?

Reply:
@franzmpub : if you see the file from the tip, https://github.com/primeqa/primeqa/blob/rag_workaround/primeqa/components/retriever/dense.py I have corrected it. Now ".search" is used only in DPR . ColBERT changes were reverted.

Reply:
@jdpsen It looks to me that line 144 is in the `ColBERTRetriever` class. 
The `p_ids, response = self._searcher.search(` call on that line has first argument `query_batch=input_texts`.